src/clj/fluree/db/constants.cljc

```diff
@@ -0,0 +1,211 @@
+(ns fluree.db.constants
+  (:require [fluree.db.json-ld.iri :as iri :refer [fluree-iri]]))
+
+#?(:clj (set! *warn-on-reflection* true))
+
+
+;; Version
+
+(def ^:const data_version 4)
+
+;; iri constants
+(def ^:const iri-CommitProof (fluree-iri "CommitProof"))
+(def ^:const iri-Commit (fluree-iri "Commit"))
+(def ^:const iri-commit (fluree-iri "commit"))
+(def ^:const iri-DB (fluree-iri "DB"))
+(def ^:const iri-data (fluree-iri "data"))
+(def ^:const iri-t (fluree-iri "t"))
+
+(def ^:const iri-address (fluree-iri "address"))
+(def ^:const iri-v (fluree-iri "v"))
+(def ^:const iri-flakes (fluree-iri "flakes"))
+(def ^:const iri-size (fluree-iri "size"))
+(def ^:const iri-assert (fluree-iri "assert"))
+(def ^:const iri-retract (fluree-iri "retract"))
+(def ^:const iri-previous (fluree-iri "previous"))
+(def ^:const iri-alias (fluree-iri "alias"))
+(def ^:const iri-ledger (fluree-iri "ledger"))
+(def ^:const iri-branch (fluree-iri "branch"))
+(def ^:const iri-issuer "https://www.w3.org/2018/credentials#issuer")
+(def ^:const iri-cred-subj "https://www.w3.org/2018/credentials#credentialSubject")
+(def ^:const iri-index (fluree-iri "index"))
+(def ^:const iri-ns (fluree-iri "ns"))
+(def ^:const iri-time (fluree-iri "time"))
+(def ^:const iri-message (fluree-iri "message"))
+(def ^:const iri-tag (fluree-iri "tag"))
+(def ^:const iri-updates (fluree-iri "updates"))
+(def ^:const iri-allow (fluree-iri "allow"))
+(def ^:const iri-equals (fluree-iri "equals"))
+(def ^:const iri-contains (fluree-iri "contains"))
+(def ^:const iri-$identity (fluree-iri "$identity"))
+(def ^:const iri-target-role (fluree-iri "targetRole"))
+(def ^:const iri-target-class (fluree-iri "targetClass"))
+(def ^:const iri-target-node (fluree-iri "targetNode"))
+(def ^:const iri-target-objects-of (fluree-iri "targetObjectsOf"))
+(def ^:const iri-property (fluree-iri "property"))
+(def ^:const iri-policy (fluree-iri "Policy"))
+(def ^:const iri-opts (fluree-iri "opts"))
+(def ^:const iri-path (fluree-iri "path"))
+(def ^:const iri-action (fluree-iri "action"))
+(def ^:const iri-all-nodes (fluree-iri "allNodes"))
+(def ^:const iri-view (fluree-iri "view"))
+(def ^:const iri-modify (fluree-iri "modify"))
+(def ^:const iri-role (fluree-iri "role"))
+(def ^:const iri-where (fluree-iri "where"))
+(def ^:const iri-values (fluree-iri "values"))
+(def ^:const iri-insert (fluree-iri "insert"))
+(def ^:const iri-delete (fluree-iri "delete"))
+
+(def ^:const iri-context "@context")
+(def ^:const iri-id "@id")
+(def ^:const iri-value "@value")
+(def ^:const iri-language "@language")
+(def ^:const iri-type "@type")
+(def ^:const iri-rdf-type "http://www.w3.org/1999/02/22-rdf-syntax-ns#type")
+(def ^:const iri-class "http://www.w3.org/2000/01/rdf-schema#Class")
+(def ^:const iri-lang-string "http://www.w3.org/1999/02/22-rdf-syntax-ns#langString")
+
+;; predicate id constants
+
+(def ^:const $_previous (iri/iri->sid iri-previous))
```

Comment:
I like that we've gotten rid of the arbitrary integer sids, I always had to be really careful not to create collisions when creating a new one.